src/libutil/strutil.cpp

```diff
@@ -524,8 +524,8 @@ Strutil::utf8_to_utf16 (string_view str)
 {
     std::wstring native;
     
-    native.resize(MultiByteToWideChar (CP_UTF8, 0, str.c_str(), -1, NULL, 0));
-    MultiByteToWideChar (CP_UTF8, 0, str.c_str(), -1, &native[0], (int)native.size());
+    native.resize(MultiByteToWideChar (CP_UTF8, 0, str.c_str(), str.length(), NULL, 0));
```

Comment:
Actually, I think that if you're passing the length explicitly, we should also pass `str.data()` rather than `str.c_str()` since if str is not already null-terminated, my `.c_str()` method will create a ustring in order to get back a null-terminated string with persistent memory.